changelogs/fragments/79017-ansible-vault-string-encryption-ending-with-newline.yml

```
@@ -0,0 +1,2 @@
+bugfixes:
+  - "ansible-vault encrypt_string - output had no newline at the end which caused broken output to stdout in some shells (e.g., trailing '%' sign in zsh). Fix now adds the newline character to the output by appending an additional element to the ``b_outs`` list (https://github.com/ansible/ansible/issues/78932)."
```

Change notes don't normally need to mention low-level implementation details. This information is useful to contributors/developers, but not the end-users. And it's already present in commits and the PR description. Changelog fragments are most useful when they explain the effect of the patch on how the end-users interface with the software, compared to how it worked in earlier versions.
How about something like this?
```suggestion
  - >-
    ansible-vault encrypt_string - started appending a line feed at the end of the encrypted string output. It was missing,
    which cased problems identifying where this string ends in some some shells (like bash) or accidentally copying an
    extra trailing terminator symbol (e.g., zsh adds prints out a ``%`` sign to signal where the original output stops)
    (https://github.com/ansible/ansible/issues/78932).
```

---

_(on an unrelated note) it's funny how a PR fixing a missing trailing LF adds a change note that does not have one :)_

---

I think I saw some detailed low-level descriptions in other changelog fragments. But it reads better without it